lib/tool-bar-view.coffee

```diff
@@ -61,6 +61,9 @@ module.exports = class ToolBarView extends View
     @.element.addEventListener 'scroll', @drawGutter
     window.addEventListener 'resize', @drawGutter
 
+    if atom.config.get 'tool-bar.automaticallyResizeIcons'
```

Comment:
Have to fully read your comment, but I see a small issue with your example code that might help you further:

```
    atom.config.observe 'tool-bar.automaticallyResizeIcons', =>
      if atom.config.get 'tool-bar.automaticallyResizeIcons'
        window.addEventListener 'resize', @updateAutoSize
...
```
The `->` needs to be an `=>` as you are referencing a method from the parent with `@`.